utils/get_prometheus_data.py

```diff
@@ -0,0 +1,114 @@
+import os
+import requests
+import json
+import logging
+import urllib3
+from datetime import datetime, timedelta
+import time
+import sys
+from prometheus_api_client import PrometheusConnect
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+logger = logging.getLogger("snafu")
+
+class get_prometheus_data():
+    def __init__(self, action):
+
+        self.uuid = action["uuid"]
+        self.user = action["user"]
+        self.cluster_name = action["cluster_name"]
+        self.test_config = action["test_config"]
+        
+        # change datetime in seconds string to datetime object
+        startime = datetime.fromtimestamp(int(action["starttime"]))
+        self.start = starttime.datetime()
+        
+        # change datetime in seconds string to datetime object
+        endtime = datetime.fromtimestamp(int(action["endtime"]))
+        # add 120s buffer to end time
+        endtime = endtime + timedelta(seconds=120)
+        self.end = endtime.datetime
+
+        # step value to be used in prometheus query
+        self.T_Delta = 30
+
+        self.get_data = False
+        if "snafu_token" in os.environ and "prometheus_url" in os.environ:
+            self.get_data = True
+            token = os.environ["snafu_token"]
+            self.url = os.environ["prometheus_url"]
+            bearer = "Bearer " + token
+            self.headers = {'Authorization': bearer}
+            self.pc = PrometheusConnect(url=self.url, headers=self.headers, disable_ssl=disable_ssl)
+        else:
+            logger.warn("""snafu service account token and prometheus url not set \n
+                        No Prometheus data will be indexed""")
+
+    def get_all_metrics(self):
+
+        # check get_data bool, if false by-pass all processing
+        if self.get_data:
+            start_time = time.time()
+
+            filename = os.path.join(sys.path[0], 'utils/prometheus_labels/included_labels.json')
+            with open(filename, 'r') as f:
+                datastore = json.load(f)
+
+            #for label in self.get_label_list():
+            for label in self.pc.all_metrics():
+
+                if label in datastore["data"]:
+                    # query_start_time = time.time()
+                    query = "irate(%s[1m])" % label
```

Comment:
based on above query, we're getting all the metrics but here were doing an irate ( which is constant ) across all metrics irrespective of the type of metric, which might not yield the result we want.

Reply:
please explain further?